### PR TITLE
Remove Dataset class in favor of type alias

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* This removes the `Dataset` class in favor of a type alias, allowing e.g. to just use a list with the appropriate fields as dataset.

```python
train_data = [
    "start": pd.Timestamp(...),
    "target": np.array([...])
]
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/model/trivial/mean.py [hunk 1]
```diff
@@ -112,12 +112,11 @@ def train(
         training_data: Dataset,
         validation_dataset: Optional[Dataset] = None,
     ) -> ConstantPredictor:
-        contexts = np.broadcast_to(
```
Comment (jaheba) [inline src/gluonts/model/trivial/mean.py]: This doesn't look like it belongs to this PR.

File: src/gluonts/evaluation/backtest.py [hunk 2]
```diff
@@ -201,7 +202,7 @@ def backtest_metrics(
     )
 
     agg_metrics, item_metrics = evaluator(
-        ts_it, forecast_it, num_series=len(test_dataset)
+        ts_it, forecast_it, num_series=maybe_len(test_dataset)
```
Comment (jaheba) [inline src/gluonts/evaluation/backtest.py]: Why is this needed?

What happens if it returns `None`?

File: src/gluonts/model/trivial/mean.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/model/trivial/mean.py]: This is needed in the PR, since `broadcast_to` requires using `len` on the `Dataset` (when it's not strictly needed), but the method may not be available being `Dataset` an `Iterable`.

File: src/gluonts/evaluation/backtest.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/evaluation/backtest.py]: Same as above: `len` may not be available. In `Evaluator.__call__` the `num_series` argument is `Optional`, and is only used for displaying progress: if a dataset has no `len`, the progress will be shown only partially (no percentage wrt total, only number of processed items).

File: src/gluonts/model/trivial/mean.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/model/trivial/mean.py]: That's mypy who asked me to do this
